packages/compass/src/app/auto-connect.ts

```diff
@@ -3,9 +3,26 @@ import { importConnections } from 'mongodb-data-service';
 import type { ConnectionInfo } from 'mongodb-data-service';
 import { promises as fsPromises } from 'fs';
 import { UUID } from 'bson';
+import { ConnectionString } from 'mongodb-connection-string-url';
+
+function applyUsernameAndPassword(
+  connectionInfo: Readonly<ConnectionInfo>,
+  { username, password }: Pick<AllPreferences, 'username' | 'password'>
+): ConnectionInfo {
+  const connectionString = new ConnectionString(connectionInfo.connectionOptions.connectionString);
+  if (username) connectionString.username = username;
```

Comment:
Currently in the username and password inputs in the connection form we `encodeURIComponent` the value that's inputted. Should we also escape these? Or better that the users do that themselves?
https://github.com/mongodb-js/compass/blob/main/packages/connection-form/src/utils/connection-string-helpers.ts#L68

Reply:
Sigh yeah, I tend to forget that `URL` doesn’t actually do this itself. Yes, we should escape these :)

Reply:
Alright, done, and also applied for `--forceConnectionOptions`